### ISSUE TITLE
Cursor position and panels reset after using clr: prefix

### ISSUE THREAD

**Issue by rohitab** [colorer/FarColorer#74]
### Far Manager version
`3.0.6157.0 x64`

### FarColorer version
`1.4.16`

### Steps to reproduce
1. Run `goto:%FARHOME%\FarEng.lng` to position the cursor on `FarEng.lng`
2. Run `clr:%FARHOME%\FarEng.lng` to open the file using FarColorer
3. Press `Esc` to close the FarColorer Viewer

### Expected behavior
Cursor position stays on the same file `FarEng.lng`.

### Actual behavior
Cursor position changes to the first item. Note that if you have a plugin panel open, like _ArcLite_ or a _Temporary Panel_, the panel will close and you will be returned to the file panel.

### Additional Information
The [`OpenW`](https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/pcolorer.cpp#L72-L100) function returns `editorSet` when called using `OPEN_COMMANDLINE`, as seen here. Due to this, the current panel is closed, and the cursor position is reset.

https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/pcolorer.cpp#L82-L84
https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/pcolorer.cpp#L99

### Possible Fix
I modified the code so that it returns the value from [`FarEditorSet::openFromCommandLine`](https://github.com/colorer/FarColorer/blob/2b42bf713ea8a22b467f5d71fbdc1ea6590beb7c/src/FarEditorSet.cpp#L1128-L1141) which is `nullptr`. After making this change and viewing a file using the `clr:` prefix, the cursor position remains the same, and plugin panels stay open.

### Patch
```patch
diff --git a/src/pcolorer.cpp b/src/pcolorer.cpp
index 517b3a3..ef479fe 100644
--- a/src/pcolorer.cpp
+++ b/src/pcolorer.cpp
@@ -80,8 +80,7 @@ HANDLE WINAPI OpenW(const struct OpenInfo* oInfo)
       editorSet->openMenu();
       break;
     case OPEN_COMMANDLINE:
-      editorSet->openFromCommandLine(oInfo);
-      break;
+      return editorSet->openFromCommandLine(oInfo);
     case OPEN_FROMMACRO:
       return editorSet->openFromMacro(oInfo);
     case OPEN_LEFTDISKMENU:

```

**Comment by rohitab** [comment]
I created a binary patch to fix this issue. I'm posting it here in case anyone is interested. The patch makes the above code changes to the plugin binary, without requiring you to build it from source.

This patch only is for FarColorer [`v1.4.16 x64`](https://github.com/colorer/FarColorer/releases/tag/v1.4.16)

### Binary Patch
Extract `FarColorer-v1.4.16.patch` from the archive [**`FarColorer-v1.4.16.patch.zip`**](https://github.com/colorer/FarColorer/files/11641814/FarColorer-v1.4.16.patch.zip) to the `%TEMP%` directory, then run the following command in `%FARHOME%\Plugins\FarColorer` directory.

```batch
git apply %TEMP%\FarColorer-v1.4.16.patch
```

**Note**: `FarColorer-v1.4.16.patch` includes the patch for #73 as well. If you want to apply the patches separately, see the following section.

### Manual Patch
Make the following changes to `bin/colorer.dll` using any Hex editor, like [Hexitor](https://sourceforge.net/projects/farplugs/files/Hexitor/).

| File Offset | Original Bytes | Patched Bytes |
|--------|--------|--------|
| `28bc4` | `8B 1D 26 00 34 00` | `31 DB 90 90 90 90` |

**Comment by ctapmex** [comment]
fixed in FarColorer v1.4.17
